compiler/ml/typecore.ml

```diff
@@ -2458,7 +2458,9 @@ and type_expect_ ?type_clash_context ?in_function ?(recarg = Rejected) env sexp
     in
     let type_clash_context = type_clash_context_from_function sexp sfunct in
     let args, ty_res, fully_applied =
-      type_application ?type_clash_context uncurried env funct sargs
```

Comment:
This somehow breaks the type environment presyumable and the following does not type check:

```res
module X = {
  type t = int
  let n : t => t = x => x+1
}

let z : X.t = 3
```

Removing the unused `let n` prevents the type error.

Reply:
Any conflicts with the existing type environment were never intended. I'll look into a fix.

However, it might add some complexity, as it is not a very elaborate solution.

Reply:
PR here: https://github.com/rescript-lang/rescript/pull/7277